spec/slather/coverage_service/html_output_spec.rb

```diff
@@ -0,0 +1,141 @@
+require File.join(File.dirname(__FILE__), '..', '..', 'spec_helper')
+require 'nokogiri'
+
+describe Slather::CoverageService::HtmlOutput do
+
+  OUTPUT_DIR_PATH = "html"
+
+  let(:fixture_html_files) do
+    ["index",
+    "fixtures.m",
+    "peekaview.m",
+    "fixtures_cpp.cpp",
+    "fixtures_mm.mm",
+    "fixtures_m.m",
+    "Branches.m",
+    "Empty.m",
+    "fixturesTests.m",
+    "peekaviewTests.m",
+    "BranchesTests.m"].map { |file| "#{file}.html"}
+  end
+
+  let(:fixtures_project) do
+    proj = Slather::Project.open(FIXTURES_PROJECT_PATH)
+    proj.extend(Slather::CoverageService::HtmlOutput)
+  end
+
+  describe '#coverage_file_class' do
+    it "should return CoverageFile" do
+      expect(fixtures_project.send(:coverage_file_class)).to eq(Slather::CoverageFile)
+    end
+  end
+
+  describe '#post' do
+    def extract_header_title(doc)
+      doc.at_css('title').text
+    end
+
+    it "should create all coverage as static html files" do
+      fixtures_project.post
+
+      fixture_html_files.map { |filename|
+        File.join(OUTPUT_DIR_PATH, filename)
+      }.each { |filepath|
+        expect(File.exists?(filepath)).to be_truthy
+      }
+
+      FileUtils.rm_rf OUTPUT_DIR_PATH if File.exists? OUTPUT_DIR_PATH
+    end
+
+    it "should create index html with correct coverage information" do
+      def extract_title(doc)
+        doc.at_css('#coverage > h2').text
+      end
+
+      def extract_coverage_text(doc)
+        doc.at_css('#total_coverage').text
+      end
+
+      def extract_coverage_class(doc)
+        doc.at_css('#total_coverage').attribute("class").to_s
+      end
+
+      def extract_cov_index(doc)
```

Comment:
I don't know how to properly compare two html files so I opted for grabbing the information that matters with nokogiri. If there's a better way (or probably a gem?) to compare html files, then please let me know.